lib/Dialect/FIRRTL/FIRRTLAttributes.cpp

```diff
@@ -122,14 +122,23 @@ StringAttr InnerSymAttr::getSymIfExists(unsigned fieldId) {
   return {};
 }
 
-StringAttr InnerSymAttr::getSymName() { return getSymIfExists(0); }
+StringAttr InnerSymAttr::getSymName() const { return getSymIfExists(0); }
 
 bool InnerSymAttr::all_of_props(
-    std::function<bool(InnerSymPropertiesAttr)> func) {
+    std::function<bool(InnerSymPropertiesAttr)> func) const {
   return llvm::all_of(getImpl()->props, func);
 }
 
-size_t InnerSymAttr::numSymbols() { return getImpl()->props.size(); }
+size_t InnerSymAttr::numSymbols() const { return getImpl()->props.size(); }
+
+bool InnerSymAttr::isSymbolInvalid() const { return getImpl()->props.empty(); }
+
+void InnerSymAttr::forAllSymNames(
```

Comment:
Should this method return bool to indicate whether a failure was encountered?

Consider LogicalResult instead of bool, which can be unclear as to which way is success (and harder to ignore by accident).  Not always better though :+1: .

Reply:
Do we need this helper function? `getSymNames()` looks sufficient to me
`llvm::all_of(inneSymAttr.getSymNames(), []())`

Reply:
Renamed it and got rid of the `getSymNames` which was just copying around the array.